api/odl.py

```diff
@@ -1518,3 +1539,38 @@ def _get(self, url, patron=None, headers=None, allowed_response_codes=None):
         self.request_args.append((patron, headers, allowed_response_codes))
         response = self.responses.pop()
         return HTTP._process_response(url, response, allowed_response_codes=allowed_response_codes)
+
+
+class ODLExpiredItemsReaper(IdentifierSweepMonitor):
+    """Responsible for removing expired ODL licenses."""
+
+    SERVICE_NAME = "ODL Expired Items Reaper"
+    PROTOCOL = ODLAPI.NAME
+
+    def __init__(self, _db, collection):
+        super(ODLExpiredItemsReaper, self).__init__(_db, collection)
+
+    def process_item(self, identifier):
+        for licensepool in identifier.licensed_through:
+            licenses_owned = licensepool.licenses_owned
+            licenses_available = licensepool.licenses_available
+
+            for license in licensepool.licenses:
+                if license.is_expired:
+                    licenses_owned -= 1
+                    licenses_available -= 1
+
+            if licenses_owned != licensepool.licenses_owned or licenses_available != licensepool.licenses_available:
+                licenses_owned = max(licenses_owned, 0)
+                licenses_available = max(licenses_available, 0)
```

Comment:
I'm concerned about this logic. What happens if a LicensePool has multiple licenses, and one of those licenses has expired but the others have not? I think this is going to clear out an entire LicensePool if only _one_ of the licenses has expired. Instead I'd expect the LicensePool's totals to be recalculated based on the licenses that _haven't_ expired.

Unlike the other stuff I've mentioned on this PR, this seems serious enough that I'd like to hear from @vbessonov and/or @tdilauro before proceeding with https://github.com/NYPL-Simplified/circulation/pull/1636.

Reply:
I was wrong about this code clearing out an entire LicensePool (I was misreading `-= 1` as `= -1`). However after  talking about this we agreed that this isn't accurately doing the math necessary to add up License values to get LicensePool values.

We don't actually know what the math _is_ -- other vendors take care of it for us -- but we have a pretty good idea.